packages/block-editor/src/components/inserter/media-tab/media-tab.js

```diff
@@ -0,0 +1,182 @@
+	rootClientId,
+	selectedCategory,
+	onSelectCategory,
+	onInsert,
+} ) {
+	const mediaCategories = useMediaCategories( rootClientId );
+	const isMobile = useViewportMatch( 'medium', '<' );
+	const baseCssClass = 'block-editor-inserter__media-tabs';
+	const onSelectMedia = useCallback(
+		( media ) => {
+			if ( ! media?.url ) {
+				return;
+			}
+			const [ block ] = getBlockAndPreviewFromMedia( media, media.type );
+			onInsert( block );
+		},
+		[ onInsert ]
+	);
+	return (
+		<>
+			{ ! isMobile && (
+				<div className={ `${ baseCssClass }-container` }>
+					<nav aria-label={ __( 'Media categories' ) }>
+						<ItemGroup role="list" className={ baseCssClass }>
+							{ mediaCategories.map( ( mediaCategory ) => (
+								<Item
+									role="listitem"
+									key={ mediaCategory.name }
+									onClick={ () =>
+										onSelectCategory( mediaCategory )
+									}
+									className={ classNames(
+										`${ baseCssClass }__media-category`,
+										{
+											'is-selected':
+												selectedCategory ===
+												mediaCategory,
+										}
+									) }
+									aria-label={ mediaCategory.label }
+									aria-current={
+										mediaCategory === selectedCategory
+											? 'true'
+											: undefined
+									}
+								>
+									<HStack>
+										<FlexBlock>
+											{ mediaCategory.label }
+										</FlexBlock>
+										<Icon icon={ chevronRight } />
+									</HStack>
+								</Item>
+							) ) }
+							<div role="listitem">
+								<MediaUploadCheck>
+									<MediaUpload
+										multiple={ false }
+										onSelect={ onSelectMedia }
+										allowedTypes={ ALLOWED_MEDIA_TYPES }
+										render={ ( { open } ) => (
+											<Button
+												onClick={ ( event ) => {
+													// Safari doesn't emit a focus event on button elements when
+													// clicked and we need to manually focus the button here.
+													// The reason is that core's Media Library modal explicitly triggers a
+													// focus event and therefore a `blur` event is triggered on a different
+													// element, which doesn't contain the `data-unstable-ignore-focus-outside-for-relatedtarget`
+													// attribute making the Inserter dialog to close.
+													event.target.focus();
+													open();
+												} }
+												className="block-editor-inserter__media-library-button"
+												variant="secondary"
+												data-unstable-ignore-focus-outside-for-relatedtarget=".media-modal"
+											>
+												{ __( 'Open Media Library' ) }
+											</Button>
+										) }
+									/>
+								</MediaUploadCheck>
+							</div>
+						</ItemGroup>
+					</nav>
+				</div>
+			) }
+			{ isMobile && (
+				<MediaTabNavigation
+					onInsert={ onInsert }
+					rootClientId={ rootClientId }
+					mediaCategories={ mediaCategories }
+				/>
+			) }
+		</>
+	);
+}
+
+function MediaTabNavigation( { onInsert, rootClientId, mediaCategories } ) {
```

Comment:
I noticed that the paddings are a bit "off" on mobile. Noting that it's also the case for the patterns tab too.

Reply:
That sounds like good small follow up, no? 😄